src/app/containers/MostRead/Canonical/index.test.jsx

```diff
@@ -1,99 +1,191 @@
 import arabicMostReadData from '#data/arabic/mostRead';
-import { service as arabicConfig } from '#app/lib/config/services/arabic';
+import pidginMostReadData from '#data/pidgin/mostRead';
+import nepaliMostReadData from '#data/nepali/mostRead';
+import { ServiceContextProvider } from '#contexts/ServiceContext';
 import {
   setStalePromoTimestamp,
   setFreshPromoTimestamp,
   setStaleLastRecordTimeStamp,
-  MostReadWithContext,
 } from '../utilities/testHelpers';
+import CanonicalMostRead from '.';
 
-const services = {
-  arabic: {
-    variant: null,
-    data: arabicMostReadData,
-    config: arabicConfig.default,
-    expectedLastUpdated: 'آخر تحديث 11 يناير/ كانون الثاني 1970',
-  },
-};
+/* eslint-disable react/prop-types */
+const MostReadCanonicalWithContext = ({
+  service,
+  endpoint,
+  initialData,
+  wrapper,
+}) => (
+  <ServiceContextProvider service={service}>
+    <CanonicalMostRead
+      endpoint={endpoint}
+      initialData={initialData}
+      wrapper={wrapper}
+    />
+  </ServiceContextProvider>
+);
 
 describe('MostReadContainerCanonical', () => {
   afterEach(() => {
     fetch.resetMocks();
   });
 
-  Object.keys(services).forEach(service => {
-    it(`should render items without timestamps for ${service}`, async () => {
-      const { variant, data: mostReadData, config } = services[service];
-      const { header, numberOfItems } = config.mostRead;
-
-      fetch.mockResponse(JSON.stringify(setFreshPromoTimestamp(mostReadData)));
+  [
+    {
+      description: 'should render 10 list items for pidgin from fetched data',
+      service: 'pidgin',
+      numberOfItems: 10,
+      mostReadData: pidginMostReadData,
+      endpoint: 'www.test.bbc.com/pidgin/mostread.json',
+    },
+    {
+      description: 'should render 5 list items for nepali from server data',
+      service: 'nepali',
+      numberOfItems: 5,
+      mostReadData: nepaliMostReadData,
+      endpoint: 'www.test.bbc.com/nepali/mostread.json',
+      initialData: nepaliMostReadData,
+    },
+  ].forEach(
+    ({
+      description,
+      service,
+      numberOfItems,
+      mostReadData,
+      endpoint,
+      initialData,
+    }) => {
+      it(description, async () => {
+        fetch.mockResponse(
+          JSON.stringify(setFreshPromoTimestamp(mostReadData)),
+        );
 
-      const { container } = render(
-        <MostReadWithContext
-          service={service}
-          variant={variant}
-          mostReadToggle
-        />,
-      );
+        let container;
+        await act(async () => {
+          container = await render(
+            <MostReadCanonicalWithContext
+              service={service}
+              endpoint={endpoint}
+              initialData={initialData}
+            />,
+          ).container;
+        });
 
-      await wait(() => {
-        expect(container.querySelector('h2').textContent).toEqual(header);
         expect(container.querySelectorAll('li').length).toEqual(numberOfItems);
```

Comment:
We can avoid this test since the following one is assuming that there also X number of `li`s

Reply:
yes good spot 👍